### PR TITLE
Allow comparison of OpaqueKeys that have the same set of fields (and the...

### PR DESCRIPTION
**Pull request by cpennington**
... same deprecation status)

### CODE REVIEW

**Comment by cpennington** [comment]
@doctoryes: Need this for https://github.com/edx/edx-platform/pull/7789

File: opaque_keys/__init__.py [hunk 1]
```diff
@@ -373,8 +373,8 @@ def __ne__(self, other):
         return not self == other
 
     def __lt__(self, other):
-        if type(self) != type(other):
-            raise TypeError('Cannot compare {} to {}.'.format(type(self), type(other)))
+        if (self.KEY_FIELDS, self.CANONICAL_NAMESPACE, self.deprecated) != (other.KEY_FIELDS, other.CANONICAL_NAMESPACE, other.deprecated):
+            raise TypeError("{!r} is incompatible with {!r}".format(self, other))
```
Comment (doctoryes) [inline opaque_keys/__init__.py]: Looks OK to me - but which OpaqueKey derivatives are you trying to compare that need this change?

Comment (cpennington) [inline opaque_keys/__init__.py]: `SlashSeparatedCourseKey` (which is deprecated, but still appears in places), and `CourseLocator`.

**Comment by doctoryes** [comment]
:+1:

**Pull request merged**